File: 1874.py
n = int(input())
st = []
t =  []
for _ in range(n):
    t.append(int(input()))

idx =0  #idx는 내가 찾아야할 수의 인덱스 번호!
ans = []
for i in range(1, n+1):
    #i는 넣을 수
    #1. 일단 넣는다!
    st.append(i)
    ans.append("+")
    #2. 아까 로직!!! 스택의 맨위와 내가 찾아야할 수가 같다면 팝!
    while st and st[-1] == t[idx]:
        st.pop()
        ans.append("-")
        idx+=1


if len(st)==0:
    print(*ans, sep="\n")
else:
    print('NO')

File: 1874.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n,idx=0;
    cin>>n;
    vector<int> st;
    vector<char> ans;
    vector<int> t;
    for(int i=0;i<n;i++){
        int temp;
        cin>>temp;
        t.push_back(temp);
    }

    for(int i=1;i<=n;i++){
        st.push_back(i);
        ans.push_back('+');
        while(!st.empty()&&st.back()==t[idx]){
            st.pop_back();
            idx+=1;
            ans.push_back('-');
        }

    }
    if(st.size()){
        cout<<"NO";
    }else{
        for(auto c : ans){
            cout<<c<<"\n";
        }
    }
}